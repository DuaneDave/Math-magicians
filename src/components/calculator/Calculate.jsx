import styles from './Calculate.module.css';
import Calculator from './Calculator';

function Calculate() {
  return (
    <div className={styles.wrapper}>
      <div className={styles.contain}>
        <p>Lets do some math!</p>
        <Calculator />
      </div>
    </div>
  );
}

export default Calculate;
