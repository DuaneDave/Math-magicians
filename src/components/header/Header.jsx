import { Link } from 'react-router-dom';

import styles from './Header.module.css';

function Header() {
  return (
    <header className={styles.header}>
      <h1 className={styles.logo}>Math Magicians</h1>
      <ul className={styles.menu}>
        <li className={styles['menu-item']}>
          <Link to="/">Home</Link>
        </li>
        <li className={styles['menu-item']}>
          <Link to="/calculate">Calculator</Link>
        </li>
        <li className={styles['menu-item']}>
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </header>
  );
}

export default Header;
