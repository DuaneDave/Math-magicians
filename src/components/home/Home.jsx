import styles from './Home.module.css';

function Home() {
  return (
    <div className={styles.home}>
      <h1>Welcome to our page!</h1>
      <p>
        Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quia deleniti
        commodi cupiditate obcaecati inventore, repellat minus facilis eligendi
        sed delectus, voluptates aliquam impedit, quaerat aperiam architecto
        nesciunt nisi recusandae rerum corrupti unde. Dolorem nulla eaque in
        quisquam, ut minus, aperiam consequatur odio laudantium iusto fugiat
        repudiandae deleniti dolores ab consectetur a mollitia eos quam natus?
        Repudiandae officiis fuga consectetur, impedit sed eligendi aperiam,
        velit, quidem hic ex perferendis culpa aut illum! Nisi, iste. Officia
        veniam quis doloribus? Officiis, accusantium nesciunt.
      </p>
    </div>
  );
}

export default Home;
