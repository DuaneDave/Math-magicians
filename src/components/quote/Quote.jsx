import styles from './Quote.module.css';

function Quote() {
  return (
    <div className={styles.quote}>
      <div className={styles.text}>
        <p>
          Mathematics is not about numbers, equations, computations or
          algorithms: it is about understanding
          <span>  -- William Paul Thurston</span>
        </p>
      </div>
    </div>
  );
}

export default Quote;
