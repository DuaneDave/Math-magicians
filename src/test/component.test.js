import React from 'react';
import renderer from 'react-test-renderer';
import { Router } from 'react-router-dom';

import Home from '../components/home/Home';
import Quote from '../components/quote/Quote';
import Calculate from '../components/calculator/Calculate';
import Header from '../components/header/Header';

describe('Component test', () => {
  it('Should render the home component', () => {
    const home = renderer.create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });

  it('Should render the quote component', () => {
    const quote = renderer.create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });

  it('Should render the calculate component', () => {
    const calculate = renderer.create(<Calculate />).toJSON();
    expect(calculate).toMatchSnapshot();
  });

  it('Should render the header component', () => {
    <Router>
      const header = renderer.create(
      <Header />
      ).toJSON(); expect(header).toMatchSnapshot();
    </Router>;
  });
});
