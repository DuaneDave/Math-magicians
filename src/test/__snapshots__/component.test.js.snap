// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test Should render the calculate component 1`] = `
<div
  className="wrapper"
>
  <div
    className="contain"
  >
    <p>
      Lets do some math!
    </p>
    <div
      className="container"
    >
      <div
        className="screen"
      />
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="btn span-2"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Component test Should render the home component 1`] = `
<div
  className="home"
>
  <h1>
    Welcome to our page!
  </h1>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Quia deleniti commodi cupiditate obcaecati inventore, repellat minus facilis eligendi sed delectus, voluptates aliquam impedit, quaerat aperiam architecto nesciunt nisi recusandae rerum corrupti unde. Dolorem nulla eaque in quisquam, ut minus, aperiam consequatur odio laudantium iusto fugiat repudiandae deleniti dolores ab consectetur a mollitia eos quam natus? Repudiandae officiis fuga consectetur, impedit sed eligendi aperiam, velit, quidem hic ex perferendis culpa aut illum! Nisi, iste. Officia veniam quis doloribus? Officiis, accusantium nesciunt.
  </p>
</div>
`;

exports[`Component test Should render the quote component 1`] = `
<div
  className="quote"
>
  <div
    className="text"
  >
    <p>
      Mathematics is not about numbers, equations, computations or algorithms: it is about understanding
      <span>
          -- William Paul Thurston
      </span>
    </p>
  </div>
</div>
`;
